src/parser/view_transform.c

```diff
@@ -3426,8 +3426,8 @@ mq_copypush_sargable_terms (PARSER_CONTEXT * parser, PT_NODE * statement, PT_NOD
       info.in.attr_list = spec->info.spec.as_attr_list;
       info.in.query_list = derived_table;
 
-      sub = derived_table->info.query.q.select.from->info.spec.derived_table;
-      if (sub && sub->node_type == PT_DBLINK_TABLE)
+      sub = derived_table->info.query.q.select.from;
+      if (sub && sub->info.spec.derived_table_type == PT_DERIVED_DBLINK_TABLE)
```

Comment:
@byungwook-kim 
It does not work properly in the following cases.
```
select .. from (select .. from db_link() a, table b ... )
```
It should be pushed like this:
```
select ..
from ( select ..
               from db_link() <== second push
             where col1 = 1 <== first push
          )
where col1 =1
```
The old source code was written to work like this.

Reply:
I tested the query as which you wrote. The rewritten query was the same as you wrote.